lib/HappyPlugin.js

```diff
@@ -162,25 +162,19 @@ HappyPlugin.prototype.start = function(compiler, done) {
 
     function resolveLoaders(callback) {
       var loaderPaths = that.state.loaders.map(function(loader) {
-        if (loader.query) {
-          return loader.path + loader.query;
-        }
-
         return loader.path;
       });
 
-      WebpackUtils.resolveLoaders(compiler, loaderPaths, function(err, loaders) {
+      WebpackUtils.resolveLoaders(compiler, loaderPaths, function(err, resolvedPaths) {
         if (err) return callback(err);
 
-        that.state.loaders = loaders.map(function(loader) {
-          if (!loader.request) {
-            loader.request = (loader.query) ? loader.path + loader.query : loader.path;
-          }
-
-          return loader;
+        resolvedPaths.forEach(function (resolvedPath, index) {
```

Comment:
I would've rather kept the previous style where we reassigned `that.state.loaders` in order to make the intent more clear, now using `forEach` it isn't as much, but it's not a big deal.